login-utils/lslogins.c

```diff
@@ -501,7 +501,8 @@ static int parse_utmpx(const char *path, size_t *nrecords, struct utmpx **record
 	if (stat(path, &st) == 0 && (size_t) st.st_size >= sizeof(struct utmpx)) {
 		imax = st.st_size / sizeof(struct utmpx);
 		ary = xreallocarray(NULL, imax, sizeof(struct utmpx));
-	}
+	} else
+		return -errno;
 
```

Comment:
There is not only `stat()` in the `if( )` :-) So, the errno will not be set if the file is empty.

Reply:
Yes, but the return value is never used and it wouldn't even make much sense to use it, as this would break the fallback to lastlog.
I think the function should be void. I'm also unsure if the `err(EXIT_FAILURE,...)` is really necessary.